lib/util.coffee

```diff
@@ -23,3 +27,73 @@ exports.treatAsMissingDevice = (uuidOrId) ->
 		replacementErr = new errors.ResinDeviceNotFound(uuidOrId)
 		replacementErr.stack = err.stack
 		throw replacementErr
+
+# Merging two sets of pine options sensibly is more complicated than it sounds.
+#
+# The general rules are:
+# * select, orderby, top and skip override (select this, instead of the default)
+# * filters are combined (i.e. both filters must match)
+# * expands are combined (include both expansions), and this recurses down.
+#   * That means $expands within expands are combined
+#   * And $selects within expands override
+# * Any unknown options 'extra' options throw an error. Unknown 'default' options are ignored.
+exports.mergePineOptions = (defaults, extras) ->
+	if not extras
+		return defaults
+
+	result = cloneDeep(defaults)
+
+	for own option, value of extras
+		switch option
+			when 'select', 'orderby', 'top', 'skip'
+				result[option] = value
+
+			when 'filter'
+				if defaults.filter
+					result.filter = $and: [defaults.filter, value]
+				else
+					result.filter = value
+
+			when 'expand'
+				result.expand = mergeExpandOptions(defaults.expand, value)
+
+			else
+				throw new Error("Unknown pine option: #{option}")
```

Comment:
It might be better to just let pinejs-client handle unknown options, that way we can automatically support new options without having to make changes here